File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/2640-maximum-number-of-integers-to-choose-from-a-range-i.java
class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        HashSet<Integer> st = new HashSet<>();
        for(int num : banned) st.add(num);

        long sum = 0;
        int count = 0;

        for(int i = 1; i<=n; i++) {
            if(!st.contains(i)) {
                sum += i;
                if(sum > maxSum) break;
                count++;
            }
        }

        return count;
    }
}

File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/2640-maximum-number-of-integers-to-choose-from-a-range-i.py
class Solution(object):
    def maxCount(self, banned, n, maxSum):
        sum, count = 0, 0
        st = set(banned)

        for i in range(1, n+1):
            if i not in st:
                sum += i
                if sum > maxSum:
                    break
                count += 1
        
        return count
        